### PR TITLE
ISSUE-21: implement opendkim.sig_geterror()

### PR DESCRIPTION
**Pull request by desttinghim**
Solves #21 
- [x] Implements opendkim.sig_geterror()
- [x] Adds documentation
- [x] Adds test for opendkim.sig_geterror()

Here I'm referring to documentation as the line in the README, though after a little more thought I guess I should write a wiki page as well. That's in progress at the time of writing.

EDIT: Created gist with documentation: https://gist.github.com/desttinghim/71ff9833fda5f198e320ec695fcc56d1

### CODE REVIEW

File: src/opendkim.cc [hunk 1]
```diff
@@ -425,8 +450,7 @@ NAN_METHOD(OpenDKIM::Chunk) {
 
   finish_chunk:
     _safe_free(&message);
-
-  info.GetReturnValue().Set(info.This());
+info.GetReturnValue().Set(info.This());
```
Comment (godsflaw) [inline src/opendkim.cc]: broke the formatting here.

File: src/opendkim.cc [hunk 2]
```diff
@@ -382,6 +383,30 @@ NAN_METHOD(OpenDKIM::SigGetSelector) {
...
+      "sig_geterror(): library must be initialized first"
+    );
+    return;
+  }
+
+  obj->GetSignature(info);
+
+  if (obj->sig == NULL) {
+    Nan::ThrowTypeError(
+      "sig_geterror(): get_error() failed, call it first for more details"
```
Comment (godsflaw) [inline src/opendkim.cc]: should read `sig_geterror(): get_signature() failed, call it first for more details`

File: test/04-verifying/29-sig-geterror.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      length: messages.good.length
+    });
+    opendkim.chunk_end();
+    var error = opendkim.sig_geterror();
+    t.is(error, 0);
+  } catch (err) {
+    console.log(err);
+    t.fail();
+  }
+});
```
Comment (godsflaw) [inline test/04-verifying/29-sig-geterror.js:27]: This test looks good, but we should add one or two more tests for known error conditions.  The same test with:
```
messages.bad_sigalg
messages.bad_signature_version
```
would tell us more about the return value from this function.  That is, that it returns a numeric code on error.

**Review by godsflaw — COMMENTED** [review]
Once the PR is ironed out, ISSUE-20 should be a walk in the park.

**Comment by godsflaw** [comment]
@desttinghim sorry for the delay on the review, I've been without power for a few days.

**Comment by desttinghim** [comment]
Okay, I've fixed the formatting errors and added two more tests. No problem with the wait.

**Comment by godsflaw** [comment]
hmm, you started working on ISSUE-20 #20 on your dev branch, so now the PR is confused.  If you can remove that last commit, I can merge this branch.

In the future, use topic branches for PRs.  You can create a branch called ISSUE-21, ISSUE-20, ISSUE-XX, etc. and then make the PR from there.  Otherwise, we will have this problem.

**Comment by desttinghim** [comment]
Oh looks like my commit for implementing issue 20 showed up here as well. Should I put that on a different branch?

**Comment by desttinghim** [comment]
Ah, sorry about that. I'll put ISSUE-20 on it's own branch then.

EDIT: Done, ISSUE-20 is now in a separate branch.

**Review by godsflaw — APPROVED** [review]

**Comment by godsflaw** [comment]
thanks @desttinghim, you can send the invoice on ethlance for the bounty, and feel free to PR ISSUE-20 #20.

**Comment by desttinghim** [comment]
Alright, thanks! Pleasure working with you.

**Pull request merged**